governator-core/src/main/java/com/netflix/governator/Governator.java

```diff
@@ -1,367 +1,339 @@
 package com.netflix.governator;
 
-import java.lang.annotation.Annotation;
-import java.lang.reflect.Method;
 import java.util.ArrayList;
 import java.util.Arrays;
 import java.util.Collection;
 import java.util.Collections;
-import java.util.HashSet;
+import java.util.IdentityHashMap;
+import java.util.LinkedHashSet;
 import java.util.List;
 import java.util.Set;
 
+import javax.inject.Inject;
+import javax.inject.Singleton;
+
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 
-import com.google.common.collect.ImmutableList;
 import com.google.inject.AbstractModule;
+import com.google.inject.ConfigurationException;
+import com.google.inject.CreationException;
 import com.google.inject.Guice;
 import com.google.inject.Injector;
-import com.google.inject.Key;
 import com.google.inject.Module;
+import com.google.inject.ProvisionException;
 import com.google.inject.Stage;
-import com.google.inject.spi.Element;
-import com.google.inject.spi.Elements;
+import com.google.inject.TypeLiteral;
 import com.google.inject.util.Modules;
-import com.netflix.governator.auto.AutoContext;
-import com.netflix.governator.auto.Condition;
-import com.netflix.governator.auto.ModuleListProvider;
-import com.netflix.governator.auto.PropertySource;
-import com.netflix.governator.auto.annotations.Conditional;
-import com.netflix.governator.auto.annotations.OverrideModule;
+import com.netflix.governator.annotations.SuppressLifecycleUninitialized;
+import com.netflix.governator.annotations.binding.Arguments;
+import com.netflix.governator.annotations.binding.Profiles;
+import com.netflix.governator.internal.DefaultPropertySource;
+import com.netflix.governator.internal.GovernatorFeatureSet;
+import com.netflix.governator.internal.ModulesEx;
+import com.netflix.governator.spi.ModuleListTransformer;
+import com.netflix.governator.spi.PropertySource;
 
 /**
- * Utility class matching Guice's {@link Guice} but providing shutdown capabilities.
- * Note that the injector being created will not by default support @PreDestory and
- * @PostConstruct.  Those are supported by adding LifecycleModule to the list of modules.
+ * Main entry point for creating a LifecycleInjector with guice extensions such as 
+ * support for @PostConstruct, @PreDestroy annotations and LifecycleListener.
  * 
- * @author elandau
- *
+ * Example:
+ * <code>
+     new Governator()
+        .addModules(
+            new ArchaiusGovernatorModule(),
+            new JettyModule(),
+            new JerseyServletModule() {
+                {@literal @}@Override
+                protected void configureServlets() {
+                    serve("/*").with(GuiceContainer.class);
+                    
+                    bind(GuiceContainer.class).asEagerSingleton();
+                    bind(HelloWorldApp.class).asEagerSingleton();
+                }  
+            }
+        )
+        .run()
+        .awaitTermination();
+ * </code>
  */
 public class Governator {
```

Comment:
This class has usage outside of our packages as far as I can tell with the Governator.createInjector style.  I don't think this is a safe change.